scripts/lean4lean_checker/main.py

```diff
@@ -0,0 +1,131 @@
+import os
+import sys
+from pathlib import Path
+import shutil
+import subprocess
+
+
+class Lean4LeanChecker():
+  def __init__(self) -> None:
+    self.project_name = "equational_theories"
+    self.l4l_repo_url = "https://github.com/digama0/lean4lean.git"
+    self.toolchain_filename = Path('lean-toolchain')
+    self.topPath = Path(os.getcwd())
```

Comment:
This is intentional

Reply:
To use a fresh copy of lean4lean built from lake from scratch. There are good reasons to avoid debugging lake updates when it can be avoided. Git clone reacts poorly with cloning into existing directories. Further a crash can leave lean projects in an inconsistent state.

Reply:
Note that getcwd() is the top level directory of the project wherever the CI runs it. All paths set up in the constructor are absolute paths.

Reply:
The script is meant to be run from the project top level. The most direct use case is the CI. Of course there is a case to be made that the same script can be used as pre-commit or pre-push hooks, we can simply link to the script from bash.

Reply:
For locally running this script I can include an instruction that says it should be run from the top level.

Reply:
This is a good idea.